src/saveload/strings_sl.cpp

```diff
@@ -123,11 +119,12 @@ struct NAMEChunkHandler : ChunkHandler {
 
 		while ((index = SlIterateArray()) != -1) {
 			if (index >= NUM_OLD_STRINGS) SlErrorCorrupt("Invalid old name index");
-			if (SlGetFieldLength() > (uint)LEN_OLD_STRINGS) SlErrorCorrupt("Invalid old name length");
+			size_t length = SlGetFieldLength();
+			if (length > LEN_OLD_STRINGS) SlErrorCorrupt("Invalid old name length");
 
-			SlCopy(&_old_name_array[LEN_OLD_STRINGS * index], SlGetFieldLength(), SLE_UINT8);
-			/* Make sure the old name is null terminated */
-			_old_name_array[LEN_OLD_STRINGS * index + LEN_OLD_STRINGS - 1] = '\0';
+			std::string &str = _old_name_array[index];
+			str.resize(length);
+			SlCopy(str.data(), length, SLE_UINT8);
```

Comment:
Feels like a helper for length + std::string would be useful in various places

Reply:
There is one other place that reads into a `std::string` in the non-oldloader code.